packages/core/src/amazonqGumby/telemetry/codeTransformMetadata.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const codeTransformMaxMetaDataLength = 65536
+
+export interface ICodeTransformMetaData {
+    dependencyVersionSelected?: string
+    canceledFromChat?: boolean
+    retryCount?: number
+    errorMessage?: string
+}
+type ICodeTransformMetaDataKeys = keyof ICodeTransformMetaData
+export const toJsonString = (metaData: ICodeTransformMetaData) => {
+    const jsonString = JSON.stringify(metaData)
+
+    if (jsonString.length <= codeTransformMaxMetaDataLength) {
+        return jsonString
+    }
+
+    let currentLength = 0
+    const objectKeys = Object.keys(metaData) as ICodeTransformMetaDataKeys[]
+    for (const key of objectKeys) {
+        const value = metaData[key]
+        // add 5 for quotes and comma around key-value pairs
+        const elementLength = key.length + JSON.stringify(value).length + 5
+        if (currentLength + elementLength <= codeTransformMaxMetaDataLength) {
+            delete metaData[key]
+            currentLength += elementLength
```

Comment:
the `truncateProps()` function may be useful https://github.com/aws/aws-toolkit-vscode/blob/c8c57b1961df9bf982d466b888549a4fc8459f56/packages/core/src/shared/utilities/textUtilities.ts#L59

Reply:
Used `partialClone` as we are removing the key for now. Not truncating it.

If the behavior changes in the future I'm leaving a comment for someone to follow up on.

Reply:
I actually kept the delete behavior. The omission of the key entirely will tell us if something was deleted and overflowed. Adding logging to users client as well.

Reply:
> The omission of the key entirely will tell us if something was deleted and overflowed.

setting the value to something like "overflow" might be useful